Release v1.15 (#181)
reductstore/website: blog/2025-05-07-reductstore-v1_15_0-released/index.mdx

---
title: ReductStore v1.15.0 Released With Extension API and Improved Web Console
description: ReductStore v1.15 introduces an extension API, improved Web Console, and new conditional query operators.
authors: alexey
tags: [news]
slug: news/reductstore-v1_15_0-released
date: 2025-05-07
image: ./img/banner_release_1_15.webp
---

![ReductStore v1.15.0 Released](./img/banner_release_1_15.webp)

> We are pleased to announce the release of the latest minor version of [**ReductStore**](/), [**1.15.0**](https://github.com/reductstore/reductstore/releases/tag/v1.15.0). ReductStore is a timeseries database designed for storing and managing large amounts of blob data.

Reviewer: ```suggestion
We are pleased to announce the release of the latest minor version of [**ReductStore**](/), [**1.15.0**](https://github.com/reductstore/reductstore/releases/tag/v1.15.0). ReductStore is a high-performance storage and streaming solution designed for storing and managing large volumes of historical data.
```

To download the latest released version, please visit our [**Download Page**](/download).

## What's new in 1.15.0?

This release includes several new features and enhancements. These are the extension API, the improved Web Console and the new conditional query operators.

{/* truncate */}

### Extension API

ReductStore is blob storage. It doesn't know anything about the data it stores.
We are determined to maintain this because it enables us to ingest and query data in any format with optimal performance.
We know that you sometimes need to perform processing and special queries based on the original data format.

For example, if you ingest data in the JSON format, you should be able to query only some fields of the JSON object or use it in the query condition for filtering.
The new extension API makes it possible.

> The extension API is experimental and not yet documented. We are developing extensions for columnar data, CSV and mcap formats.

Reviewer: ```suggestion
The extension API is experimental and not yet documented. We are developing extensions for columnar data, CSV and MCAP formats.
```

Once we have enough experience, we will document the API and publish the extensions so that you can build your own extensions for your data formats.

For most curious users, a demo extension that scales JPEG images on the fly can be found on GitHub: https://github.com/reductstore/img-ext

### Improved Web Console

In the v1.14.0 release, we introduced the ability to browse data in the Web Console.
This release includes two new features: the ability to upload files to the database and update labels in the Web Console.

![Update Labels in ReductStore Web Console](./img/web-console.png)

> The upload feature can be useful when you store some artifacts e.g. AI models or configuration files in the storage and want to update it somtimes.

Reviewer: ```suggestion
The upload feature can be useful when you store some artifacts e.g. AI models or configuration files in the storage and want to update it occasionally.
```
